controllers/searchoperator_controller.go

```diff
@@ -495,12 +501,20 @@ func updateOperatorCR(kclient client.Client, cr *searchv1alpha1.SearchOperator,
 		return err
 	}
 	cr.Status.PersistenceStatus = status
+	if cr.Status.PreviousState == nil {
+		cr.Status.PreviousState = &deploy
+	} else if cr.Status.PreviousState != nil {
+		if *cr.Status.PreviousState == deploy {
+			updState := !deploy
+			cr.Status.PreviousState = &updState
+		}
+	}
```

Comment:
Maybe this can be simpler. We can just always set "PreviousState" to `deployVarPresent`.

Reply:
Corrected, thought we want to persist the previous state.